es714x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/SqlIndicesEsRequestContext.scala

```diff
@@ -67,42 +70,86 @@ class SqlIndicesEsRequestContext private(actionRequest: ActionRequest with Compo
       case Right(sqlIndices) =>
         val indicesStrings = indices.map(_.stringify).toList.toSet
         if (indicesStrings != sqlIndices.indices) {
-          SqlRequestHelper.modifyIndicesOf(request, sqlIndices, indicesStrings) match {
+          val result = for {
+            _ <- SqlRequestHelper.modifyIndicesOf(request, sqlIndices, indicesStrings)
+            _ <- Right(applyFieldLevelSecurityTo(request, fieldLevelSecurity))
+            _ <- Right(applyFilterTo(request, filter))
+          } yield ()
+          result match {
             case Right(_) =>
-              applyFieldLevelSecurity(request, fieldLevelSecurity)
-              Modified
+              UpdateResponse { response =>
+                Task.delay {
+                  applyFieldLevelSecurityTo(response, fieldLevelSecurity) match {
+                    case Right(modifiedResponse) => modifiedResponse
+                    case Left(SqlRequestHelper.ModificationError.UnexpectedException(ex)) =>
+                      // todo: log & better ex
+                      throw new IllegalStateException("dd")
+                  }
+                }
+              }
             case Left(SqlRequestHelper.ModificationError.UnexpectedException(ex)) =>
               logger.error(s"[${id.show}] Cannot modify SQL indices of incoming request", ex)
               CannotModify
           }
         } else {
-          applyFieldLevelSecurity(request, fieldLevelSecurity)
-          Modified
+          applyFieldLevelSecurityTo(request, fieldLevelSecurity)
+          applyFilterTo(request, filter)
+
+          UpdateResponse { response =>
+            Task.delay {
+              applyFieldLevelSecurityTo(response, fieldLevelSecurity) match {
+                case Right(modifiedResponse) => modifiedResponse
+                case Left(SqlRequestHelper.ModificationError.UnexpectedException(ex)) =>
+                  // todo: log & better ex
+                  throw new IllegalStateException("dd")
```

Comment:
what does "dd" mean?

Reply:
Nevermind, it's not  in final version. Probably some polish acronym lol.

Reply:
`dd` - vim's remove line ;)